backend/data/src/main/java/ru/matcher/data/model/converter/ConverterOrganizationType.java

```diff
@@ -0,0 +1,27 @@
+package ru.matcher.data.model.converter;
+
+import ru.matcher.commons.OrganizationType;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import java.util.Objects;
+
+/**
+ * Конвертер типа организации
+ *
+ * @author Максим Щербаков
+ */
+@Converter(autoApply = true)
+public class ConverterOrganizationType implements AttributeConverter<OrganizationType, Integer> {
+
+    @Override
+    public Integer convertToDatabaseColumn(OrganizationType attribute) {
+        Objects.requireNonNull(attribute, "OrganizationType cannot be null");
+        return attribute.getId();
+    }
+
+    @Override
+    public OrganizationType convertToEntityAttribute(Integer dbData) {
+        return OrganizationType.fromId(dbData).orElse(null);
```

Comment:
 Лучше `orElseThrow()`. Тогда будет явно падать эксепшен о том, что значения нет.

Reply:
Мы это давно ещё решили, так как не нашли подходящего эксепшена

Reply:
Из сурсов:
```java 
/**
     * If a value is present, returns the value, otherwise throws
     * {@code NoSuchElementException}.
     *
     * @return the non-{@code null} value described by this {@code Optional}
     * @throws NoSuchElementException if no value is present
     * @since 10
     */
    public T orElseThrow() {
        if (value == null) {
            throw new NoSuchElementException("No value present");
        }
        return value;
    }
```
Он по умолчанию кидает вполне правильный эксепшен. Это лучше, чем потом бд поругается, что ему почему-то `null` в `NOT NULL` поле прилетел.